src/txn/txn_timestamp.c

```diff
@@ -459,6 +464,50 @@ __wt_txn_global_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
 	return (0);
 }
 
+#ifdef HAVE_TIMESTAMPS
+/*
+ * __wt_txn_commit_ts_validate --
+ *	Validate commit timestamp being set
+ */
+int
+__wt_txn_commit_ts_validate(WT_SESSION_IMPL *session,
```

Comment:
The other timestamp APIs all use `timestamp` not `ts` in their names. i.e: `__wt_txn_commit_timestamp_validate`

Reply:
Can this be extended to include validation of other timestamps?  How about `wt_timestamp_validate`?  I can see that validating read timestamps is similar.  Maybe pass in a flag such as read or commit for the non-overlapping parts.